### PR TITLE
seeed-xiao-rp2040: Add initial board support

### PR DESCRIPTION
**Pull request by rcsim**
## Summary
Initial board support for Seeed Studio Xiao RP2040

## Impact
User will be able to use Nuttx on Seeed Studio Xiao RP2040

## Testing
Currently only nsh defconfig is supported:
- make distclean
- ./tools/configure.sh -l seeed-xiao-rp2040:nsh
- make 
- Boot board in  "BOOT mode"and copy nuttx.uf2 into the device.
- Reset the board and connect to UART0
- User should be able to see NSH
image

### CODE REVIEW

**Comment by acassis** [comment]
@xiaoxiang781216 any idea what is the error?

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt [hunk 1]
```diff
@@ -0,0 +1,100 @@
+README
+======
```
Comment (xiaoxiang781216) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:2]: move to Documentation folder

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  - Persistent flash filesystem in unused flash ROM
+
+NuttX also provide support for these external devices:
+
+  - BMP180 sensor at I2C0 (don't forget to define I2C0 GPIOs at "I2C0 GPIO pin assign" in Board Selection menu)
+  - INA219 sensor / module (don't forget to define I2C0 GPIOs at "I2C0 GPIO pin assign" in Board Selection menu)
+  - Pico Display Pack (ST7789 LCD)
+    - RGB leds and buttons are not supported yet.
+  - Pico Audio Pack (PCM5100A I2S DAC)
+    - I2S interface is realized by PIO.
```
Comment (acassis) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:34]: I think xiao-rp2040 form-factor is not compatible with Pico Display Pack or Pico Audio Pack, please remove!

Comment (rcsim) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt]: Hi @acassis ,
I removed this info.
Thanks!

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt (hunk 1 above)
Comment (rcsim) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:2]: Hi @xiaoxiang781216 , 
This README.txt file is already on Documentation folder (Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt), so I`m not sure what I need to do here, could you please let me know ?
Thanks

Comment (xiaoxiang781216) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:2]: Sorry, I mean change to .rst format, but I saw many README is still in text format. So, please ignore my comment.

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt [hunk 3]
```diff
@@ -0,0 +1,96 @@
+README
+======
+
+This directory contains the port of NuttX to the Seeed Studio Xiao RP2040.
+See ttps://wiki.seeedstudio.com/XIAO-RP2040/ for information about Seeed
```
Comment (hartmannathan) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:5]:
```suggestion
See https://wiki.seeedstudio.com/XIAO-RP2040/ for information about Seeed
```

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    - MSC, CDC/ACM serial and these composite device are supported.
+    - CDC/ACM serial device can be used for the console.
+  - PIO (RP2040 Programmable I/O)
+  - Flash ROM Boot
+  - SRAM Boot
+    - If Pico SDK is available, nuttx.uf2 file which can be used in
+      BOOTSEL mode will be created.
+  - Persistent flash filesystem in unused flash ROM
+
+NuttX also provide support for these external devices:
```
Comment (hartmannathan) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:27]:
```suggestion
NuttX also provides support for these external devices:
```

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+==========
+
+- nsh
+    Minimum configuration with NuttShell
+
+License exceptions
+==================
+
+The following files are originated from the files in Pico SDK.
+So, the files are licensed under 3-Clause BSD same as Pico SDK.
```
Comment (hartmannathan) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt:83]: The listed files and their license need to be added to LICENSE in the top-level directory.

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+- A NeoPixel RGB LED connected to GPIO11 (NEOPIXEL_POWER) and GPIO12 (PIN_NEOPIXEL).
+
+Buttons
+=======
+
+There are 2 buttons available:
+
+A RESET button and a BOOT button, which if held down when power is first
+applied to the board, will cause the RP2040 to boot into programming
+mode and appear as a storage device to a computer connected via USB .
```
Comment (hartmannathan) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst:45]:
```suggestion
mode and appear as a storage device to a computer connected via USB.
```

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+11    3V3        Power output to peripherals
+12    Ground
+13    VIN        +5V Supply to board
+===== ========== ==========
+
+Power Supply
+============
+For general I/O pins:
+
+Working voltage of MCU is 3.3V . Voltage input connected to general I/O pins
```
Comment (hartmannathan) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst:77]:
```suggestion
Working voltage of MCU is 3.3V. Voltage input connected to general I/O pins
```

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+may cause chip damage if it' higher than 3.3V .
```
Comment (hartmannathan) [inline Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst:78]:
```suggestion
may cause chip damage if it' higher than 3.3V.
```

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 it seems like our CI is broken again!

 Fixed here: https://github.com/apache/nuttx/pull/11282

**Comment by xiaoxiang781216** [comment]
please rebase to the last master which fix the recent ci break

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt (hunk 3 above)
Comment (rcsim) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt]: Hi @hartmannathan ,
It has been fixed.
Thanks

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt (hunk 4 above)
Comment (rcsim) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/README.txt]: Hi @hartmannathan ,
It has been fixed.
Thanks

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst (hunk 6 above)
Comment (rcsim) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst]: Hi @hartmannathan ,
It has been fixed.
Thanks

File: Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst (hunk 8 above)
Comment (rcsim) [reply Documentation/platforms/arm/rp2040/boards/seeed-xiao-rp2040/index.rst]: Hi @hartmannathan ,
It has been fixed.
Thanks

**Review by hartmannathan — APPROVED** [review]
LGTM, thanks! We'll wait for CI checks to complete, hopefully with success.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**